libnavui-status/src/main/java/com/mapbox/navigation/ui/status/view/MapboxStatusView.kt

```diff
@@ -0,0 +1,238 @@
+package com.mapbox.navigation.ui.status.view
+
+import android.animation.Animator
+import android.animation.AnimatorInflater
+import android.animation.AnimatorSet
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import android.widget.ProgressBar
+import androidx.annotation.AnimatorRes
+import androidx.annotation.UiThread
+import androidx.appcompat.widget.AppCompatImageView
+import androidx.appcompat.widget.AppCompatTextView
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.animation.doOnStart
+import androidx.core.view.isInvisible
+import androidx.core.view.isVisible
+import com.mapbox.navigation.ui.status.R
+import com.mapbox.navigation.ui.status.databinding.MapboxStatusViewLayoutBinding
+import com.mapbox.navigation.ui.status.internal.extensions.doOnFinish
+import com.mapbox.navigation.ui.status.model.Status
+
+/**
+ * View for rendering [Status] information.
+ */
+@Suppress("MemberVisibilityCanBePrivate")
+class MapboxStatusView : FrameLayout {
+
+    private val binding = MapboxStatusViewLayoutBinding.inflate(LayoutInflater.from(context), this)
+    private var pendingHideAnimation: Animator? = null
+
+    /**
+     * Currently rendered [Status].
+     */
+    var currentStatus: Status? = null
+        private set
+
+    /**
+     * Container view that hosts [spinnerProgressBar], [iconImage] and [messageTextView].
+     */
+    val containerView: ConstraintLayout = binding.container
+
+    /**
+     * Indeterminate Progress indicator displayed when [Status.spinner] is `true`.
+     */
+    val spinnerProgressBar: ProgressBar = binding.progressBar
+
+    /**
+     * Icon Image displayed when [Status.icon] is set.
+     */
+    val iconImage: AppCompatImageView = binding.iconImage
+
+    /**
+     * TextView used to display [Status.message].
+     */
+    val messageTextView: AppCompatTextView = binding.messageText
+
+    /**
+     * Returns `true` if this view is visible.
+     */
+    val isRendered: Boolean get() = isVisible
+
+    /**
+     * A resource identifier for the Show Animator.
+     */
+    @AnimatorRes
+    var showAnimRes: Int = 0
+
+    /**
+     * A resource identifier for the Hide Animator.
+     */
+    @AnimatorRes
+    var hideAnimRes: Int = 0
+
+    constructor(context: Context) : this(context, null)
```

Comment:
must ba called `super`, not `this`, here and below

Reply:
[All android views](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget) are using `this()` to call lower level constructor.

Reply:
Yeah, for instance we can take [EditText](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/widget/EditText.java). If we don't use `this`, then lose "default" style `com.android.internal.R.attr.editTextStyle` for constructor: 

```java
public EditText(Context context, AttributeSet attrs) {
 this(context, attrs, com.android.internal.R.attr.editTextStyle);
}
```
To prevent this, we should always call `this()`(same reason we also don't use `@JvmOverload`)